test/conformance/conformancetest_helper.go

```diff
@@ -31,18 +32,33 @@ import (
 	. "github.com/knative/serving/pkg/reconciler/testing"
 )
 
-// fetchRuntimeInfo creates a Service that uses the 'runtime' test image, and extracts the returned output into the
+// The 'runtime-unprivileged' test image uses uid 1000.
+const unprivilegedUserID = 1000
+
+// fetchRuntimeInfoUnprivileged creates a Service that uses the 'runtime-unprivileged' test image, and extracts the returned output into the
 // RuntimeInfo object.
+func fetchRuntimeInfoUnprivileged(t *testing.T, clients *test.Clients, options *test.Options, opts ...ServiceOption) (*test.ResourceNames, *types.RuntimeInfo, error) {
+	return runtimeInfo(t, clients, &test.ResourceNames{Image: runtimeUnprivileged}, options, opts...)
+}
+
+// fetchRuntimeInfo creates a Service that uses the 'runtime' test image, and extracts the returned output into the
+// RuntimeInfo object. The 'runtime' image uses uid 0.
 func fetchRuntimeInfo(t *testing.T, clients *test.Clients, options *test.Options, opts ...ServiceOption) (*test.ResourceNames, *types.RuntimeInfo, error) {
-	names := test.ResourceNames{
-		Service: test.ObjectNameForTest(t),
-		Image:   runtime,
+	return runtimeInfo(t, clients, &test.ResourceNames{}, options, opts...)
+}
+
+func runtimeInfo(t *testing.T, clients *test.Clients, names *test.ResourceNames, options *test.Options, opts ...ServiceOption) (*test.ResourceNames, *types.RuntimeInfo, error) {
+	names.Service = test.ObjectNameForTest(t)
+	if names.Image == "" {
+		names.Image = runtime
+	} else if names.Image != runtimeUnprivileged {
```

Comment:
why this restriction?

Reply:
The function won't work unless the service returns the expected runtime image response. I put it there mostly for easy debugging if an unexpected image is passed in.

I thought about adding an unprivileged bool/enum rather than using the resource names, but it felt worse.